File: AEDS2/provas/1/q2/q1.java
import java.util.Scanner;

public class q1 {
  public static class Research {
    private int rabbit;
    private int rat;
    private int frog;

    Research() {
      this.rabbit = 0;
      this.rat = 0;
      this.frog = 0;
    }

    public void addRabbit(int r) { this.rabbit += r; }
    public void addRat(int r) { this.rat += r; }
    public void addFrog(int f) { this.frog += f; }

    public int getRabbit() { return this.rabbit; }
    public int getRat() { return this.rat; }
    public int getFrog() { return this.frog; }
    public int getTotal() { return this.rat + this.frog + this.rabbit; }
  }  

  public static double getPercentage(int count, int total) {
    return (double) (count * 100/ (double) total);
  }

  public static void main(String[] args) {
      Scanner input = new Scanner(System.in);
      int n = input.nextInt();
      input.nextLine();
      Research re = new Research(); 
      String line;
      for (int i = 0; i < n; i++) {
        line = input.nextLine();
        String[] temp = line.split("\\s");
        int animals = Integer.parseInt(temp[0]);
        if (temp[1].charAt(0) == 'C') re.addRabbit(animals);
        else if (temp[1].charAt(0) == 'R') re.addRat(animals);
        else if (temp[1].charAt(0) == 'S') re.addFrog(animals);
      }

      System.out.println("Total: " + re.getTotal() + " cobaias");
      System.out.println("Total de coelhos: " + re.getRabbit());
      System.out.println("Total de ratos: " + re.getRat());
      System.out.println("Total de sapos: " + re.getFrog());
      System.out.printf("Percentual de coelhos: %.2f %%\n", getPercentage(re.getRabbit(), re.getTotal()));
      System.out.printf("Percentual de ratos: %.2f %%\n", getPercentage(re.getRat(), re.getTotal()));
      System.out.printf("Percentual de sapos: %.2f %%\n", getPercentage(re.getFrog(), re.getTotal()));
      input.close();
  }
}

File: AEDS2/provas/1/q2/q1.c
#include <stdio.h>

typedef enum {
  RABBIT,
  RAT,
  FROG
} species;

typedef struct {
  species specie;
  int count;
} animal;

typedef struct {
  animal rabbit;
  animal rat;
  animal frog;
  int total;
} research;

void init_research(research* r) {
  // not really used here, but is a good idea in general
  r->rabbit.specie = RABBIT;
  r->rat.specie    = RAT;
  r->frog.specie   = FROG;

  r->rabbit.count = 0;
  r->rat.count = 0;
  r->frog.count = 0;
  r->total = 0;
}

float get_percentage(animal a, int total) {
  return a.count * 100 / (float) total;
}

int main () {

  int r_count;
  if (scanf("%d", &r_count) != 1) return -1;

  research re;
  init_research(&re);
  
  for (int i = 0; i < r_count; i++) {
    int num;
    char specie;
    if(scanf("%d %c", &num, &specie) != 2) perror("failed to read line\n");

    switch(specie) {
      case 'C':
        re.rabbit.count += num;
        break;
      case 'R':
        re.rat.count += num;
        break;
      case 'S':
        re.frog.count += num;
        break;
      default:  
        perror("shouldnt be reached.");
        break;
    }

    re.total += num;
  }

  printf("Total: %d cobaias\n", re.total);
  printf("Total de coelhos: %d\n", re.rabbit.count);
  printf("Total de ratos: %d\n", re.rat.count);
  printf("Total de sapos: %d\n", re.frog.count);
  printf("Percentual de coelhos: %.2f %%\n", get_percentage(re.rabbit, re.total));
  printf("Percentual de ratos: %.2f %%\n", get_percentage(re.rat, re.total));
  printf("Percentual de sapos: %.2f %%\n", get_percentage(re.frog, re.total));
  
  return 0;
}
